File: src/Back-End/Question.cs
using System;
using System.Collections.Generic;

//Object that contains the type of question as an enum, the question itself, and the answer
public class Question
{
    public enum QuestionType { MajorTalkingPoint, MinorTalkingPoint, SharedExperiences, SocialBarriers };
    public int numberOfAnswersAllowed;
    public QuestionType Type;
    public int QuestionId;
    public string QuestionText;
    public Dictionary<string, List<AnswerRelationship>> Answers;

    public Question(QuestionType type, string questionText, Dictionary<string, List<AnswerRelationship>> answer, int numberOfAnswersAllowed = 1)
    {
        Type = type;
        QuestionText = questionText;
        Answer = answer;
        MultipleAnswers = multipleAnswers;
        //TODO: Add QuestionId

    }

    private class AnswerRelationship {
        public string key;
        public int value;

        public AnswerRelationship(string key, int value) {
            this.key = key;
            this.value = value;
        }
    }
}

File: src/Back-End/question.ts
export interface Question {
    numberOfAnswersAllowed: number;
    type: QuestionType;
    questionId: number;
    questionText: string;
    //Key: answer text, Value: { Key: other answers Value: relationshipScore}
    answers: { [key: string]: {[key: string]: number; }; };
}

export enum QuestionType {
    MajorTalkingPoint,
    MinorTalkingPoint,
    SharedExperiences,
    SocialBarriers
}

export function getWeightOfQuestionType(type: QuestionType): number {
    switch (type) {
        case QuestionType.MajorTalkingPoint:
            return 0.2;
        case QuestionType.MinorTalkingPoint:
            return 0.1;
        case QuestionType.SharedExperiences:
            return 0.3;
        case QuestionType.SocialBarriers:
            return 0.4;
    }
}

export class Question {
    numberOfAnswersAllowed: number;
    type: QuestionType;
    questionId: number;
    questionText: string;
    //Key: answer text, Value: { Key: other answers Value: relationshipScore}
    answers: { [key: string]: {[key: string]: number; }; };

    constructor(numberOfAnswersAllowed: number, type: QuestionType, questionId: number, questionText: string) {
        this.numberOfAnswersAllowed = numberOfAnswersAllowed;
        this.type = type;
        this.questionId = questionId;
        this.questionText = questionText;
        this.answers = {};
    }
    

    public addAnswer(answer: string): void {
        this.answers[answer] = {};
    }

    public addRelationshipScore(answer: string, otherAnswer: string, relationshipScore: number): void {
        this.answers[answer][otherAnswer] = relationshipScore;
    }

    public getRelationshipScore(answer: string, otherAnswer: string): number {
        return this.answers[answer][otherAnswer];
    }
}

